tests/sentry/issues/test_priority.py

```diff
@@ -0,0 +1,31 @@
+from sentry.issues.priority import PriorityChangeReason, PriorityLevel, auto_update_priority
+from sentry.models.group import GroupStatus
+from sentry.testutils.cases import TestCase
+from sentry.testutils.helpers.features import apply_feature_flag_on_cls
+
+
+@apply_feature_flag_on_cls("projects:issue-priority")
+class TestUpdatesPriority(TestCase):
+    def test_updates_priority_escalating(self) -> None:
```

Comment:
Could we add a test to ensure that priority is not changed when `priority_locked_at` is set?

We also should probably test that the Activity and GroupHistory entries are created correctly